### PR TITLE
First draft of 3p type specific inclusion guidelines.

### PR DESCRIPTION
**Pull request by cramforce**

### CODE REVIEW

File: 3p/3P_CONTRIBUTIONS.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Inclusion of third party software into AMP
```
Comment (jmadler) [inline 3p/3P_CONTRIBUTIONS.md]: We should probably clarify this is for plugins for third-party services, not just third-party software

File: 3p/3P_CONTRIBUTIONS.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+- Iframes and all sub resources must be served from HTTPS.
+- Avoid client side rendering of iframe content.
+- If your use of iframes is for style isolation, consider that AMP might provide an iframe-free alternative.
+- If you can make it not iframe based that is much better. (See e.g. the pinterest embed). We will always ask to do this first. E.g. adding a CORS endpoint to your server might make this possible.
+- Must play well within AMP's sizing framework.
+- All JS on container page must be open source and bundled with AMP.
+- Direct iframe embeds not using our 3p iframe mechanism (used e.g. for ads) are preferred.
+- JavaScript loaded into iframe should be reasonable with respect to functionality.
+- Use the `sandbox` attribute on iframe if possible.
+- Embeds that require plugins such as Flash or Java applets need to go through special review.
```
Comment (jmadler) [inline 3p/3P_CONTRIBUTIONS.md]: We should probably word a bit stronger: "Embeds that require browser plugins, such as Flash, Java, ActiveX, Silverlight, etc. are disallowed unless necessary"

File: 3p/3P_CONTRIBUTIONS.md (hunk 1 above)
Comment (cramforce) [inline 3p/3P_CONTRIBUTIONS.md]: Done

File: 3p/3P_CONTRIBUTIONS.md (hunk 2 above)
Comment (cramforce) [inline 3p/3P_CONTRIBUTIONS.md]: Done

File: 3p/3P_CONTRIBUTIONS.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+- We welcome pull requests by all ad networks for inclusion into AMP.
+- All ads and all sub resources must be served from HTTPS.
+- Must play well within AMP's sizing framework.
+- Direct iframe embeds not using our 3p iframe mechanism (used by most ads) are preferred.
+- Providing an image only zero-iframe embed is appreciated.
+- Support viewability and other metrics/instrumentation as supplied by AMP (via postMessage API)
+- Try to keep overall iframe count at one per ad. Explain why more are needed.
+- Share JS between iframes on the same page.
+
+The following aren't hard requirements, but please present a road map for when this can be achieved:
```
Comment (jmadler) [inline 3p/3P_CONTRIBUTIONS.md]: Suggest being a bit more explicit, and softer on sharing the roadmap:

The following aren't hard requirements, but are performance optimizations we should strive to incorporate.  If possible, please provide a timeline as to when you expect to follow these guidelines:

**Comment by cramforce** [comment]
PTAL

**Comment by jmadler** [comment]
One last note above, LGTM afterwards

**Comment by jmadler** [comment]
Should this be 3p/README.md?

**Comment by cramforce** [comment]
Yeah, renamed. Can diversify to more files if needed later.

**Pull request merged**